pkg/macvlan/macvlan_test.go

```diff
@@ -0,0 +1,134 @@
+package macvlan
+
+import (
+	"fmt"
+	"testing"
+
+	egresstest "github.com/openshift/egress-router-cni/pkg/testing"
+	netlink_mocks "github.com/openshift/egress-router-cni/pkg/testing/mocks/github.com/vishvananda/netlink"
+	util "github.com/openshift/egress-router-cni/pkg/util"
+	util_mocks "github.com/openshift/egress-router-cni/pkg/util/mocks"
+	"github.com/vishvananda/netlink"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestFillNetConfDefaults(t *testing.T) {
+	mockNetLinkOps := new(util_mocks.NetLinkOps)
+	mockLink := new(netlink_mocks.Link)
+	// below sets the `netLinkOps` in util/net_linux.go to a mock instance for purpose of unit tests execution
+	util.SetNetLinkOpMockInst(mockNetLinkOps)
+
+	tests := []struct {
+		desc             string
+		inpNetConf       *NetConf
```

Comment:
this is always filled with a zero-ed struct, why not moving it inside the test body?
It would reduce the cluttering

Reply:
+1

Reply:
I had missed a test earlier, In my latest push of the PR (https://github.com/openshift/egress-router-cni/pull/31) I make use of this field